File: 1290-convert-binary-number-in-a-linked-list-to-integer/1290-convert-binary-number-in-a-linked-list-to-integer.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int getDecimalValue(ListNode* head) {
        return solve(head);
    }
private:
    int solve(ListNode* head){
        int ans = 0;
        ListNode* temp = head;
        
        // Traverse all the nodes
        while(temp != nullptr){
            
            // adding the node value at last bit of answer
            ans = (ans << 1) | (temp -> val);
            temp = temp -> next;
        }
        return ans;
    }
};

File: 1290-convert-binary-number-in-a-linked-list-to-integer/1290-convert-binary-number-in-a-linked-list-to-integer.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def getDecimalValue(self, head: ListNode) -> int:
        ans = 0
        
        temp = head
        
        # Traverse all nodes of list
        while temp != None:
            
            # Add the bit at the last position of ans bit
            ans = (ans << 1) | temp.val
            temp = temp.next
        return ans